model/delivery/container/LtiDeliveryContainer.php

```diff
@@ -24,29 +24,57 @@
 use oat\taoDelivery\model\container\execution\ExecutionClientContainer;
 use oat\taoDelivery\model\container\ExecutionContainer;
 use oat\taoDelivery\model\execution\DeliveryExecution;
+use IMSGlobal\LTI\ToolProvider\ToolConsumer;
+use oat\oatbox\session\SessionService;
+use oat\generis\model\OntologyAwareTrait;
+use oat\tao\model\oauth\DataStore;
 
 /**
  * Class LtiDeliveryContainer
  *
  * A delivery container to manage LTI based delivery
- *
- * @package oat\taoLtiConsumer\model\delivery\container
  */
 class LtiDeliveryContainer extends AbstractContainer
 {
+    use OntologyAwareTrait;
+
     /**
      * Get the execution container to render LTI based delivery
      *
      * @param DeliveryExecution $execution
+     *
      * @return ExecutionClientContainer|ExecutionContainer
-     * @throws \common_exception_NotImplemented
+     * @throws \common_exception_InvalidArgumentType
+     * @throws \common_exception_NotFound
      */
     public function getExecutionContainer(DeliveryExecution $execution)
     {
-        throw new \common_exception_NotImplemented('LTI delivery execution container is not yet implemented.');
-        $container = new ExecutionClientContainer($execution);
-        $container->setData('deliveryExecution', $execution->getIdentifier());
-        $container->setData('deliveryServerConfig', []);
+        $params = $this->getRuntimeParams();
+        $providerResource = $this->getResource($params['ltiProvider']);
+        $ltiUrl = $params['ltiPath'];
+        $ltiProvider = $providerResource->getPropertiesValues([
+            DataStore::PROPERTY_OAUTH_KEY,
+            DataStore::PROPERTY_OAUTH_SECRET,
+            DataStore::PROPERTY_OAUTH_CALLBACK,
+        ]);
+        $consumerKey = (string)reset($ltiProvider[DataStore::PROPERTY_OAUTH_KEY]);
```

Comment:
do you really need to reset ?

Reply:
This looks like java engineer code... right? ;-)
Elements are indexed with a predicate string, so to get the first element, we could change `reset` with `current` but it doesn't really improve the readability...
I'd leave it as is.

Reply:
ok, did not see the multi dimensional approach, then ok :)